File: Challenge4/JavaScript/Challenge4.js
function Palindromic() {
    var highestPalindromic = 0;
    for (var i = 0; i < 1000; i++) {
        for (var j = 0; j < 1000;
            j++) {

            var totalNumber = i * j
            var numberReversed = parseInt((("" + totalNumber).split("")).reverse().join(""));

            if (totalNumber == numberReversed && highestPalindromic < numberReversed) {
                highestPalindromic = totalNumber;
            }
        }
    }
    return highestPalindromic
}

console.log(Palindromic());

File: Challenge4/Java/Challenge4.java
public class Challenge4
{   
        public static void main(String[] args){
            System.out.println(Palindromic());
        }

        public static int Palindromic()
        {
            int highestPalindromic = 0;
            for (int i = 0; i < 1000; i++)
            {
                for (int j = 0; j < 1000; j++)
                {
                    int totalNumber = i * j;
                    char[] totalChar = ("" + totalNumber).toCharArray();
                    char[] charReversed = new StringBuilder(new String(totalChar)).reverse().toString().toCharArray();
											
					int reversedNumber = Integer.parseInt(new String(charReversed));

                    if (totalNumber == reversedNumber && highestPalindromic < totalNumber)
                    {
                        highestPalindromic = totalNumber;
                    }
                }
            }
            return highestPalindromic;
        }    
}
